resources/middleware.py

```diff
@@ -0,0 +1,111 @@
+import urllib3
+from starlette.datastructures import URL, QueryParams
+from starlette.requests import Request
+import json
+import os
+
+# Middleware adds the CloudFront alternative hostname (e.g. map-tiles.princeton.edu)
+# to a header so that TiTiler can generate TileJSON documents with the correct url.
+class HostMiddleware:
+    def __init__(self, app):
+        self.app = app
+
+    async def __call__(self, scope, receive, send):
+        if scope["type"] not in (
+          "http"
+        ):
+            await self.app(scope, receive, send)
+            return
+
+        headers = dict(scope["headers"])
+        headers[b"host"] = bytes(os.getenv("TITILER_BASE_URL"), 'utf-8')
+        scope["headers"] = [(k, v) for k, v in headers.items()]
+
+        await self.app(scope, receive, send)
+
+# Middleware to rewrite requests URLs before passing to TiTiler.
+# https://map-tiles.princeton.edu/{id}/{service} ->
+# https://map-tiles.princeton.edu/{service}?url=s3://{bucket}/{path}
+#
+# Examples:
+#   https://map-tiles.princeton.edu/1234567/mosaicjson?
+#   https://map-tiles.princeton.edu/mosaicjson?url=s3://figgy-geo-production/12/34/56/1234567/mosaic.json
+#
+#   https://map-tiles.princeton.edu/1234567/cog?id=1234567 ->
```

Comment:
delete this query string at the end of this?

Reply:
Done.